### PR TITLE
chore: improve `Ecosystem.md` linter to lint all sections

### PR DESCRIPTION
**Pull request by nooreldeensalah**
Extend the ecosystem linter functionality (module name validation + ordering) to all `Ecosystem.md` sections (`Core`, `Community`, `Community Tools`) instead of `Community` only.

Relevant Issues/PRs: #4256 #4257

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

File: .github/scripts/lint-ecosystem.js [hunk 1]
```diff
@@ -19,15 +19,29 @@ module.exports = async function ({ core }) {
   let inCommunitySection = false
   let modules = []
   let hasImproperFormat = false
+  let moduleName
 
   for await (const line of rl) {
     lineNumber += 1
+
+    if (line.startsWith('- [') === true) {
```
Comment (jsumners) [inline .github/scripts/lint-ecosystem.js:27]: We can cache this `startsWith` since it is used again later.

**Review by jsumners — APPROVED** [review]
Looks good to me.

File: .github/scripts/lint-ecosystem.js (hunk 1 above)
Comment (nooreldeensalah) [reply .github/scripts/lint-ecosystem.js]: Thank you for your suggestion, I've added a commit to address this change.
I've also added two more commits:
- One commit to extend the functionality of the ordering to all sections.
- Another commit to fix out of order modules in the `Core` section.
If you feel like the last two commits are irrelevant to this PR, or not required, I can drop them if you wish

**Comment by nooreldeensalah** [comment]
I've added a few commits to extend the functionality to all sections (ordering + validation), and I've fixed a few out of order modules in the `Core` section

**Comment by Eomm** [comment]
Could you merge `main`?
CI is red because of #4260

**Pull request merged**